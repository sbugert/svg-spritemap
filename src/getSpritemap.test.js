const getSpritemap = require('./getSpritemap');

test('generates svg spritemap for files in directory', async () => {
  expect(
    await getSpritemap('./fixtures', { optimize: true })
  ).toMatchSnapshot();
});

test('omits svg tag', async () => {
  expect(
    await getSpritemap('./fixtures', { withSvgTag: false, optimize: true })
  ).toMatchSnapshot();
});

test('does not insert new title if old exists', async () => {
  expect(await getSpritemap('./fixtures/3')).toMatchSnapshot();
});
