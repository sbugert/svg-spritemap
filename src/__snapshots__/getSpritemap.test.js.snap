// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not insert new title if old exists 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" style=\\"display: none;\\"><defs><symbol width=\\"300\\" height=\\"200\\" baseProfile=\\"full\\" id=\\"svg-test3\\"><title>test 3 title</title><rect width=\\"100%\\" height=\\"100%\\"/></symbol></defs></svg>"`;

exports[`generates svg spritemap for files in directory 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" style=\\"display: none;\\"><defs><symbol width=\\"300\\" height=\\"200\\" baseProfile=\\"full\\" id=\\"svg-test\\"><title>test</title><rect width=\\"100%\\" height=\\"100%\\"/><circle cx=\\"150\\" cy=\\"100\\" r=\\"80\\"/><text x=\\"150\\" y=\\"125\\" font-size=\\"60\\" text-anchor=\\"middle\\">SVG</text></symbol><symbol width=\\"300\\" height=\\"200\\" baseProfile=\\"full\\" id=\\"svg-test2\\"><title>test2</title><rect width=\\"100%\\" height=\\"100%\\"/></symbol></defs></svg>"`;

exports[`omits svg tag 1`] = `"<defs><symbol width=\\"300\\" height=\\"200\\" baseProfile=\\"full\\" id=\\"svg-test\\"><title>test</title><rect width=\\"100%\\" height=\\"100%\\"/><circle cx=\\"150\\" cy=\\"100\\" r=\\"80\\"/><text x=\\"150\\" y=\\"125\\" font-size=\\"60\\" text-anchor=\\"middle\\">SVG</text></symbol><symbol width=\\"300\\" height=\\"200\\" baseProfile=\\"full\\" id=\\"svg-test2\\"><title>test2</title><rect width=\\"100%\\" height=\\"100%\\"/></symbol></defs>"`;
